index.js

```diff
@@ -152,23 +189,24 @@ var diff = function(input, output, path=[], diffMatchPatch) {
 		return ops;
 	} else if (Array.isArray(output) || Array.isArray(input)) {
 		// deal with array/object
-		var op = { p: path };
-		op[isObject ? "od" : "ld"] = input;
-		op[isObject ? "oi" : "li"] = output;
-		return [op];
+		return replaceOp(path, isObject, input, output, json1);
 	}
 
 	var ops = [];
 	var keys = new Set([...Object.keys(input), ...Object.keys(output)]);
 	keys.forEach(function(key) {
-		var newOps = diff(input[key], output[key], [...path, key], diffMatchPatch);
+		var newOps = diff(input[key], output[key], [...path, key], options);
 		ops = ops.concat(newOps);
 	});
 	return ops;
 }
 
-var optimizedDiff = function(input, output, diffMatchPatch) {
-	return optimize(diff(input, output, [], diffMatchPatch));
+var optimizedDiff = function(input, output, options) {
+	if (options && !options.diffMatchPatch) {
```

Comment:
(Moving this conversation here)

>Indeed, it's a bit awkward to accept either `diffMatchPatch` or an options object. Perhaps it would be best to modify it such that instead of an options object, the additional options were simply more arguments?

Yes, I think so. I don't see us adding so many more arguments that it's going to become a problem. In the current state, I'd be afraid of breaking somebody's code at least.